File: 30-Day LeetCoding Challenge/C++/checkValidString.cpp
class Solution {
public:
    bool checkValidString(string s) {
        int sum = 0;
        for(int i = 0; i < s.length(); i++) {
            if(s[i] == '(' || s[i] == '*') {
                sum++;
            }
            else {
                sum--;
                if(sum == -1) return false;
            }
        }
        sum = 0;
        for(int i = s.length()-1; i > -1; i--) {
            if(s[i] == ')' || s[i] == '*') {
                    sum++;
            }
            else {
                sum--;
                if(sum == -1) return false;
            }
        }
        return true;
    }
};

File: 30-Day LeetCoding Challenge/Python/checkValidString.py
class Solution(object):
    def checkValidString(self, s):
        """
        :type s: str
        :rtype: bool
        """
        sum = 0
        for i in range(len(s)):
            if s[i] == '(' or s[i] == '*':
                sum += 1
            else:
                sum -= 1
                if(sum == -1):
                    return False
        sum = 0
        for i in range(len(s)-1,-1,-1):
            if s[i] == ')' or s[i] == '*':
                sum += 1
            else:
                sum -= 1
                if(sum == -1):
                    return False
        return True